lib/Gateway.js

```diff
@@ -1592,8 +1595,22 @@ Gateway.prototype.discoverValue = function (node, vId) {
         }
 
         cfg = copy(hassCfg.sensor_generic)
-        cfg.object_id =
-          sensor.sensor + (sensor.objectId ? '_' + sensor.objectId : '')
+        // Assemble an object ID including propertyKey and property.
+        // PropertyKey is set before property as propertyKey can have multiple relevant properties.
+        if (valueId.propertyKey) {
+          cfg.object_id = [
+            sensor.sensor,
+            sensor.objectId,
+            valueId.propertyKey,
+            valueId.property
+          ]
+            .filter(Boolean)
```

Comment:
Instead of doing this twice, do it only after the if-else

Reply:
to reduce the confusion i think this is a better approach:
```
cfg.object_id =
  sensor.sensor + (sensor.objectId ? '_' + sensor.objectId : '')
if (valueId.propertyKey) {
  cfg.object_id = 
    cfg.object_id +
    (valueId.propertyKey ? '_' + valueId.propertyKey : '') +
    (valueId.property ? '_' + valueId.property : '')
}
```

Reply:
Ok :+1:

Reply:
pushed fixes for this and the filter proposal on previous